SU2_CFD/include/objectives/turbomachinery/CTurbomachineryPerformance.hpp

```diff
@@ -0,0 +1,254 @@
+
+  CTurbomachineryState GetOutletState() const { return OutletState; }
+
+  su2double GetKineticEnergyLoss() const { return KineticEnergyLoss; }
+
+  su2double GetTotalPressureLoss() const { return TotalPressureLoss; }
+
+  su2double GetEntropyGen() const { return EntropyGen; }
+
+  su2double GetPressureRatio() const { return PressureRatio; }
+
+  su2double GetEulerianWork() const { return EulerianWork; }
+
+};
+
+class CTurbineBladePerformance : public CTurbomachineryBladePerformance {
+
+  public:
+  CTurbineBladePerformance(CFluidModel *const fluidModel, unsigned short nDim, su2double areaIn, su2double radiusIn,
+                           su2double areaOut, su2double radiusOut);
+
+  ~CTurbineBladePerformance();
+
+  void ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) override;
+
+};
+
+class CCompressorBladePerformance : public CTurbomachineryBladePerformance {
+
+  public:
+  CCompressorBladePerformance(CFluidModel *const fluidModel, unsigned short nDim, su2double areaIn, su2double radiusIn,
+                              su2double areaOut, su2double radiusOut);
+
+  ~CCompressorBladePerformance();
+
+  void ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) override;
+
+};
+// class CTurbomachineryStagePerformance :  CTurbomachineryBladePerformance {
+//    protected: 
+//       su2double TotalStaticEfficiency, TotalTotalEfficiency, KineticEnergyLoss, TotalPressureLoss, EntropyGen, PressureRatio, EulerianWork;
+//    public:
+//       CTurbomachineryStagePerformance();
+//       ~CTurbomachineryStagePerformance();
+//       virtual void ComputePerformance(CFluidModel* fluidModel, su2double tangVel);
+//       su2double GetTotalStaticEfficiency() const { return TotalStaticEfficiency; }
+//       su2double GetTotalTotalEfficiency() const { return TotalTotalEfficiency; }
+// };
+
+// class CTurbineStagePerformance : CTurbomachineryStagePerformance {
+
+//    public:
+//       CTurbineStagePerformance();
+//       ~CTurbineStagePerformance();
+//       void ComputePerformance(CFluidModel* fluidModel, su2double tangVel) override;
+
+// };
+
+// class CCompressorStagePerformance : CTurbomachineryStagePerformance {
+
+//    public:
+//       CCompressorStagePerformance();
+//       ~CCompressorStagePerformance();
+//       void ComputePerformance(CFluidModel* fluidModel, su2double tangVel) override;
+
+// };
+
+class CTurbomachineryPerformance {
+  private:
+  vector <vector<shared_ptr < CTurbomachineryBladePerformance>>>
+  BladesPerformances;
+
+  static void ComputePerBlade(vector <shared_ptr<CTurbomachineryBladePerformance>> const bladePerformances,
+                              vector <CTurbomachineryCombinedPrimitiveStates> const bladePrimitives);
+
+  static void ComputePerSpan(shared_ptr <CTurbomachineryBladePerformance> const spanPerformances,
+                             const CTurbomachineryCombinedPrimitiveStates &spanPrimitives);
```

Comment:
Try to be consistent with where you place `const`, and if in the bottom function spanPrimitives is passed by reference I suppose a vector of them (bladePrimitives in top function) should also be passed by reference).